### ISSUE TITLE
[Issue] Something's up with the Chameleon skin infusion, still

### ISSUE THREAD

**Issue by Pivonomicon** [GTNewHorizons/ThaumicHorizons#58]
The chameleon skin infusion keeps turning you invisible, just less frequently than before the recent patches. 
Sorry if you are already aware of the issue, i'm incredibly excited to see this mod get updated, so happy you are working on it! :D

**Comment by Alastors** [comment]
Yeah that's weird, thank you for the ticket!

**Comment by Pivonomicon** [comment]
Of course! When i apply the infusion i can toggle the invisibility on and off, no problems there, but when i try to walk around while visible it sometimes turns me invisible on it's own, no chat message indicating i'm invisible now or anything. It seems irregular, when this occurs, sometimes i walk for over half a minute, other times it's every 3 seconds. I'm trying to recreate this on a clean installation with nothing but thaumcraft and thaumic horizons right now

**Comment by Alastors** [comment]
That's really strange, keep me updated on how the redo goes!

**Comment by Alastors** [comment]
I think the sneak is supposed to happen, but the only reason I can think for the sprint is that maybe it's supposed to de-toggle the effect when you run? That's really strange

**Comment by Pivonomicon** [comment]
This was driving me insane for the last year, but, now that the mod is cared for once more, perhaps this issue might be resolved? In any case i'm looking forward to any and all updates :D

**Comment by Alastors** [comment]
Yeah! of course! 

I'll look into it as soon as possible.

And me too! 😂

**Comment by Pivonomicon** [comment]
That's so great to hear, you all are the best!
Before i stop bothering you, can i ask about a certain bug in base Thaumcraft 4 with Advanced Alchemical Furnace?

**Comment by Alastors** [comment]
Yeah of course! And you're not bothering me, I promise 😂

**Comment by Pivonomicon** [comment]
So, there is a lighting bug that was driving me absolutely crazy in Tc4, when alchemical furnace is within rendering distance all decorative blocks' shadows become bugged, they swap depending on what angle the player is looking, i have no clue how to fix it, all i know is that it occurs when there is a working, powered Advanced alchemical furnace in the loaded chunks.

![y51qh87kzyc91](https://github.com/GTNewHorizons/ThaumicHorizons/assets/144031213/e8dc9256-53ad-4f8c-a3dd-0176b8438881)
![8i4zq67kzyc91](https://github.com/GTNewHorizons/ThaumicHorizons/assets/144031213/24296e30-15cd-4529-9ba2-76bcb88e7fdf)

**Comment by Alastors** [comment]
> So, there is a lighting bug that was driving me absolutely crazy in Tc4, when alchemical furnace is within rendering distance all decorative blocks' shadows become bugged, they swap depending on what angle the player is looking, i have no clue how to fix it, all i know is that it occurs when there is a working, powered Advanced alchemical furnace in the loaded chunks.
> 
> ![y51qh87kzyc91](https://user-images.githubusercontent.com/144031213/265486618-e8dc9256-53ad-4f8c-a3dd-0176b8438881.png) ![8i4zq67kzyc91](https://user-images.githubusercontent.com/144031213/265486632-24296e30-15cd-4529-9ba2-76bcb88e7fdf.png)

Have you tried TC4Tweaks? It fixes a few bugs like that?  (Assuming you’re not from GTNH anyway, if you are then Lmfao)

**Comment by Alastors** [comment]
If TC4Tweaks doesn’t fix the bug, then try reporting it here: https://github.com/GTNewHorizons/NewHorizons/issues and I’ll ping Glee about it (they’re the one who handles all the mixins related to base Thaum and whatnot)

**Comment by Pivonomicon** [comment]
Thanks a lot! I'll be sure to do that!

**Comment by BB-621** [comment]
Very sad to report that the Chameleon skin bug persists in the exact same form as it did when reported here for the first time. Playing on version 1.7.3 (current latest), after performing the Chameleon skin infusion on a body, or using the creative item Self: Chameleon skin, the effect works as intended, being turned on and off by the keybind, (default X), and showing the appropriate chat message,  but as soon as I crouch by hitting shift, or sprint, either by double-tapping W, or by hitting Ctrl, the player immediately turns invisible. The player must then hit X (or the respective keybind for toggle) until the chat message for Chameleon skin being disabled pops up, so that the player is once again visible. I tested this on a survival world, and a creative one as well, with the same result.

Additionally, this bug also works the same on version 1.6.2-pre. This ability has been bugged for the longest time, and it's a shame, because it could be really fun and useful, but being invisible all the time is just annoying.
